### PR TITLE
Only add libintvector.h include when necessary for mozilla dist

### PR DESCRIPTION
**Pull request by franziskuskiefer**
I'd recommend doing something like this in general. But this is a quick fix for the mozilla C code.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -781,6 +781,14 @@ dist/ccf/Makefile.basic: POLY_BUNDLE =
 #
 # Disable the EverCrypt and MerkleTree layers. Only keep Chacha20, Poly1305,
 # Curve25519 for now. Everything else in Hacl is disabled.
+dist/mozilla/Makefile.basic: INTRINSIC_FLAGS =
```
Comment (s-zanella) [inline Makefile]:
```suggestion
dist/mozilla/Makefile.basic: INTRINSIC_FLAGS = \
  -add-include 'Hacl_Chacha20Poly1305_128:"libintvector.h"' \
  -add-include 'Hacl_Chacha20Poly1305_256:"libintvector.h"' \
  -add-include 'Hacl_Chacha20_Vec128:"libintvector.h"' \
  -add-include 'Hacl_Chacha20_Vec256:"libintvector.h"' \
  -add-include 'Hacl_Poly1305_128:"libintvector.h"' \
  -add-include 'Hacl_Poly1305_256:"libintvector.h"'
```

File: Makefile [hunk 2]
```diff
@@ -781,6 +781,14 @@ dist/ccf/Makefile.basic: POLY_BUNDLE =
...
+dist/mozilla/Makefile.basic: DEFAULT_FLAGS += \
```
Comment (s-zanella) [inline Makefile]:
```suggestion
```

File: Makefile [hunk 3]
```diff
@@ -781,6 +781,14 @@ dist/ccf/Makefile.basic: POLY_BUNDLE =
...
+  -add-include 'Hacl_Chacha20Poly1305_128:"libintvector.h"' \
```
Comment (s-zanella) [inline Makefile]:
```suggestion
```

File: Makefile [hunk 4]
```diff
@@ -781,6 +781,14 @@ dist/ccf/Makefile.basic: POLY_BUNDLE =
...
+  -add-include 'Hacl_Chacha20Poly1305_256:"libintvector.h"' \
```
Comment (s-zanella) [inline Makefile]:
```suggestion
```

File: Makefile [hunk 5]
```diff
@@ -781,6 +781,14 @@ dist/ccf/Makefile.basic: POLY_BUNDLE =
...
+  -add-include 'Hacl_Chacha20_Vec128:"libintvector.h"' \
```
Comment (s-zanella) [inline Makefile]:
```suggestion
```

File: Makefile [hunk 6]
```diff
@@ -781,6 +781,14 @@ dist/ccf/Makefile.basic: POLY_BUNDLE =
...
+  -add-include 'Hacl_Chacha20_Vec256:"libintvector.h"' \
```
Comment (s-zanella) [inline Makefile]:
```suggestion
```

File: Makefile [hunk 7]
```diff
@@ -781,6 +781,14 @@ dist/ccf/Makefile.basic: POLY_BUNDLE =
...
+  -add-include 'Hacl_Poly1305_128:"libintvector.h"' \
```
Comment (s-zanella) [inline Makefile]:
```suggestion
```

File: Makefile [hunk 8]
```diff
@@ -781,6 +781,14 @@ dist/ccf/Makefile.basic: POLY_BUNDLE =
...
+  -add-include 'Hacl_Poly1305_256:"libintvector.h"'
```
Comment (s-zanella) [inline Makefile]:
```suggestion
```

**Comment by franziskuskiefer** [comment]
> 
> 
> Just a nit: can you pass the flags in `INTRINSIC_FLAGS` instead, like in my comments?
> (Sorry, GitHub doesn't support multi-line suggestions yet)

Sure, like this? 🙂

**Comment by s-zanella** [comment]
Exactly. Thanks!
I queued a build since the trigger for PRs is buggy. I will merge once it finishes.

**Pull request merged**